zenpy/lib/endpoint.py

```diff
@@ -365,6 +365,22 @@ def __call__(self, **kwargs):
                 break
         return Url(endpoint_path, params=params)
 
+class TalkEndpoint(BaseEndpoint):
```

Comment:
You should be able to use the PrimaryEndpoint and SecondaryEndpoint classes for these. Let me know if you have any questions about this.

Reply:
Yes I see that now, I will should be able to get rid of my endpoint class and use those instead. Seems pretty straight forward. I should get some time soon to make those changes.

Reply:
So I have most of this working with your suggestions, however the only hiccup that I notice is that Primary endpoint will not work with the kwargs that I need unless I modify it.

one example, "phone_number_ids" for current queue activity, is not present in PrimaryEndpoint and because there is no code like :

else: params = kwargs (this would just be an example)

that feeds through the kwargs if necessary, my kwargs just don't get fed through, do I modify the Primary endpoint to take into account things other than just just what shows up in all of your elif's or do I keep my own endpoint classes? Or perhaps there is something I am just not seeing?

Thanks for your help, almost done.

Joshua

Reply:
Yep should be fine to add this to the PrimaryEndpoint. It's not ideal, and if I could go back in time I wouldn't have that god class but the horse has already bolted on that one :)

Reply:
Ok, I will add an else statement which will make sure the kwargs make it into params. Thanks for your help.